gatsby-node.js

```diff
@@ -44,3 +45,16 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
     })
   })
 }
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions
+
+  if (node.internal.type === `MarkdownRemark`) {
+    const value = createFilePath({ node, getNode })
+    createNodeField({
+      name: `slug`,
+      node,
+      value,
+    })
```

Comment:
This is slightly confusing. Does `create` mean different things in these two, or does `createFilePath` actually create a file?

Meantime, I wonder why [this](https://github.com/keep-network/tbtc-website/commit/71dd10dd83061079ce8632adab20fea054238397) isn't doing the trick 🤔

Reply:
>This is slightly confusing
Agreed!

I don't think [createFilePath](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/#createfilepath) actually creates the file, but just generates a customizable path?

As I was writing a conjecture of why that commit didn't do the trick, I realized the conjecture doesn't make sense. 😅 I thought maybe it's to do with node vs page, but that commit adds it to the node section. I took this logic from [how to create slugs for pages](https://www.gatsbyjs.org/tutorial/part-seven/#creating-slugs-for-pages) from the gatsby tutorial.

Reply:
Ok, I didn't think so. But `createNodeField` dispatches an *action* that does update state. And `createFileNodeFromBuffer` *will* create something on the filesystem. That is truly awful lol.

Still puzzled as to why the other thing doesn't work, but that's fine. Two things related to that:
- Maybe we remove that old fix? I think we can just revert that commit.
- What's an example of a news item that wasn't working? So I can double-check that the change landed as expected :)

Reply:
`createNodeField` vs `createFileNodeFromBuffer` Oh boy, yes that is awful. 🙃 

I'll test out reverting the old commit locally now, just in case.

The missing news items are these entries:
src/pages/news/2020-04-06-tbtc-will-launch-on-april-27-2020.md
src/pages/news/2020-03-25-tbtc-is-now-fully-open-sourced.md

Reply:
Hmm, it didn't like the revert. It complained:
```

 ERROR #11323 

Your site's "gatsby-node.js" must set the page path when creating a page.

The page object passed to createPage:
{
    "path": null,
    "component": "/Users/erinng/Dev/tbtc-website/src/templates/news-item.js",
    "context": {
        "id": "2f9710cd-f2cd-59d0-83a8-6ebf26022008"
    }
}

See the documentation for the "createPage" action — https://www.gatsbyjs.org/docs/actions/#createPage
```

Looks like the fix is still needed to handle null paths

Reply:
But... 🤔

Reply:
These must be hitting the same problem at different parts of the pipeline? One for actually rendering the page proper (e.g. https://tbtc.network/news/2020-04-06-tbtc-will-launch-on-april-27-2020) and one for its place in the collection.

Reply:
Just realized though that means we're taking different approaches to the two… We probably want to use the same thing in `slug`-land for the bit in `createPages`.

Reply:
Must be. Kind of maddening. 😠

Reply:
> We probably want to use the same thing in slug-land for the bit in createPages

Yes, that would be better!

Reply:
> 😠

I was feeling more 🤬🤬🤬 tbh :p

Reply:
🤣🤣🤣

Reply:
Okay 🐌'd it